### PR TITLE
added join-room & leave-room socket messages

### PR DESCRIPTION
**Pull request by CaseyMichael**

### CODE REVIEW

**Comment by t-laird** [comment]
looks good

File: client/src/App.js [hunk 1]
```diff
@@ -8,20 +8,46 @@ class App extends Component {
   constructor() {
     super();
     this.state = {
-      formValue: ''
+      formValue: '',
+      roomId: 0
     };
-    this.socket = io();
+    this.socket = io('http://localhost:5000');
```
Comment (CaseyMichael) [inline client/src/App.js]: connect to local websocket on port 5000

File: client/src/App.js [hunk 2]
```diff
@@ -8,20 +8,46 @@ class App extends Component {
...
 
-    this.socket.on('message', message => {
+    this.socket.on('play', roomId => {
```
Comment (CaseyMichael) [inline client/src/App.js]: client to list for the play message from the server

File: server.js [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
 const io = require('socket.io')(http);
-
+const winston = require('winston');
 const path = require('path');
-
 const environment = process.env.NODE_ENV || 'development';
 const config = require('./knexfile')[environment];
 const database = require('knex')(config);
 
+const logger = winston.createLogger({
```
Comment (CaseyMichael) [inline server.js:13]: good old winston!

File: server.js [hunk 4]
```diff
@@ -19,48 +28,27 @@ app.use(bodyParser.urlencoded({ extended: true }));
...
-      message: message.msg,
-      user_id: message.user.id
-    };
-    database('messages').insert(msg)
-      .then(() => {
-      })
-      .catch(err => {
-        console.log(`Error adding message: ${err}`);
-      });
+  socket.on('join-room', roomId => {
```
Comment (CaseyMichael) [inline server.js:35]: yes it really is this simple to join and leave rooms

File: client/src/Components/Controls.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import io from 'socket.io-client';
+import Url from 'url-parse';
+
+class Controls extends Component {
```
Comment (CaseyMichael) [inline client/src/Components/Controls.js:4]: Saw you created components, so I moved everything I did into a controls component but I dont know how to get this to be displayed....

File: client/src/App.js [hunk 6]
```diff
@@ -1,30 +1,11 @@
...
+import React, { Component} from 'react';
 import { Switch, Route } from 'react-router';
 import { withRouter } from 'react-router';
 import './App.scss';
-import io from 'socket.io-client';
 import Room from './Components/Room';
 import Home from './Components/Home';
 
 class App extends Component {
-  constructor() {
```
Comment (CaseyMichael) [inline client/src/App.js:10]: none of this was needed.

File: client/src/Components/Controls.js (hunk 5 above)
Comment (CaseyMichael) [reply client/src/Components/Controls.js:4]: figured this out it was actually pretty similar to how angular works.

File: client/src/Components/Home.js [hunk 7]
```diff
@@ -19,11 +20,12 @@ class Home extends Component {
     render() {
         return (
             <div className="home">
+                <Controls></Controls>
                 <div className="form">
                     <input type="text" value={this.state.formValue} onChange={this.handleChange} placeholder="enter youtube URL" />
                     <button onClick={this.handleSubmit}>Submit</button>
                 </div>
-                <div class="yt-embedded-player">
```
Comment (CaseyMichael) [inline client/src/Components/Home.js:26]: this was throwing console logs so i changed it.

**Review by t-laird — APPROVED** [review]
noice

**Pull request merged**